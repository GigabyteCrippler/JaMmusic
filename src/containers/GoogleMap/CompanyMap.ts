import faker from 'faker';

class CompanyMap {
  name: string;

  catchPhrase: string;

  loc: { lat: number, lng: number };

  constructor() {
    this.name = faker.company.companyName();
    this.catchPhrase = faker.company.catchPhrase();
    this.loc = {
      lat: parseFloat(faker.address.latitude()),
      lng: parseFloat(faker.address.longitude()),
    };
  }
}

export default CompanyMap;
