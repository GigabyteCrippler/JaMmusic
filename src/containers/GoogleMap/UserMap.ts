import faker from 'faker';

class UserMap {
  name: string;

  loc: { lat: number, lng: number };

  constructor() {
    this.name = faker.name.firstName();
    this.loc = {
      lat: parseFloat(faker.address.latitude()),
      lng: parseFloat(faker.address.longitude()),
    };
  }
}

export default UserMap;
