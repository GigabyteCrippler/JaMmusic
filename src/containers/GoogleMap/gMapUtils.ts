import { Loc } from './gMapTypes';

export interface IgMapUtils {
  makeLoc: any; limitLat: any
}
function makeLoc(faker: any): Loc {
  return {
    lat: parseFloat(faker.address.latitude()),
    lng: parseFloat(faker.address.longitude()),
  };
}
function limitLat(obj: any): any {
  const newObj = obj;
  if (obj.loc.lat > 83) newObj.loc.lat = 83;
  if (obj.loc.lat < -70) newObj.loc.lat = -70;
  return newObj;
}

export default { makeLoc, limitLat };
