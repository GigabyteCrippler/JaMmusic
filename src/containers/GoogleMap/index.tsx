import React, { useEffect } from 'react';
import UserMap from './UserMap';
import CompanyMap from './CompanyMap';

const GoogleMap = (): JSX.Element => {
  useEffect(() => {
    const gMap = new google.maps.Map(document.getElementById('googleMap'), { zoom: 2, center: { lat: 0, lng: 0 } });
    console.log(gMap);
  });
  const userMap = new UserMap();
  console.log(userMap);
  const companyMap = new CompanyMap();
  console.log(companyMap);
  return (
    <div className="page-content">
      <div style={{ margin: 'auto', textAlign: 'center' }}>
        <h2 style={{ marginTop: '10px' }}>Google Map Here</h2>
        <div id="googleMap" />
      </div>
    </div>
  );
};
export default GoogleMap;
