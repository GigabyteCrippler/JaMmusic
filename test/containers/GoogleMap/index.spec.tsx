
import * as React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import GoogleMap from '../../../src/containers/GoogleMap';

describe('GoogleMap container', () => {
  let wrapper: ShallowWrapper<Readonly<any> & Readonly<{ children?: React.ReactNode; }>, Readonly<any>, GoogleMap>;
  const fakeFunc: any = jest.fn();
  beforeEach(() => {
    global.google = {
      maps: {
        ...window.google.maps,
        Marker: fakeFunc,
        Map: fakeFunc,
      },
    };
    wrapper = shallow<GoogleMap>(<GoogleMap />);
  });
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
