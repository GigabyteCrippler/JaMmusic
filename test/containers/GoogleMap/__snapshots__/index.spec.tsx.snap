// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleMap container renders correctly 1`] = `undefined`;
